jupyter_console/interactiveshell.py

```diff
@@ -220,6 +231,47 @@ def handle_execute_reply(self, msg_id, timeout=None):
 
             self.execution_count = int(content["execution_count"] + 1)
 
+    def handle_is_complete_reply(self, msg_id, timeout=None):
+        """
+        Wait for a repsonse from the kernel, and return two values:
+            more? - (boolean) should the frontend ask for more input
+            indent - an indent string to prefix the input
+        Overloaded methods may want to examine the comeplete source. Its is
+        in the self._source_lines_buffered list.
+        """
+        ## Get the is_complete response:
+        msg = None
+        count = 0
+        max_tries = 5
+        while self.client.is_alive() and count < max_tries:
```

Comment:
I don't quite like this loop. For one thing, it means that the actual timeout is 5 times the value you pass in for `timeout`. But also, since we're using a blocking client, there shouldn't be any other messages coming back on the shell socket before your reply. If there are, something has gone wrong, and as this will consume and discard messages, it may obscure information useful for finding out what actually happened.

I think we should call `get_msg` just once, without the loop, and have a loud warning if it's something other than our reply.